test/integration/second-opinion.ts

```diff
@@ -0,0 +1,195 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
+
+import { HardhatEthersSigner } from "@nomicfoundation/hardhat-ethers/signers";
+
+import { SecondOpinionOracleMock } from "typechain-types";
+
+import { ether, impersonate, log, ONE_GWEI } from "lib";
+import { getProtocolContext, ProtocolContext } from "lib/protocol";
+import { finalizeWithdrawalQueue, norEnsureOperators, report, sdvtEnsureOperators } from "lib/protocol/helpers";
+
+import { bailOnFailure, Snapshot } from "test/suite";
+
+const AMOUNT = ether("100");
+const MAX_DEPOSIT = 150n;
+const CURATED_MODULE_ID = 1n;
+const INITIAL_REPORTED_BALANCE = ether("32") * 3n; // 32 ETH * 3 validators
+
+const ZERO_HASH = new Uint8Array(32).fill(0);
+
+// Diff amount is 10% of total supply
+function getDiffAmount(totalSupply: bigint): bigint {
+  return (totalSupply / 10n / ONE_GWEI) * ONE_GWEI;
+}
+
+describe("Second opinion", () => {
+  let ctx: ProtocolContext;
+
+  let ethHolder: HardhatEthersSigner;
+  let stEthHolder: HardhatEthersSigner;
+
+  let snapshot: string;
+  let originalState: string;
+
+  let secondOpinion: SecondOpinionOracleMock;
+  let totalSupply: bigint;
+
+  before(async () => {
+    ctx = await getProtocolContext();
+
+    [stEthHolder, ethHolder] = await ethers.getSigners();
+
+    snapshot = await Snapshot.take();
+
+    const { lido, depositSecurityModule, oracleReportSanityChecker } = ctx.contracts;
+
+    await finalizeWithdrawalQueue(ctx, stEthHolder, ethHolder);
+
+    await norEnsureOperators(ctx, 3n, 5n);
+    await sdvtEnsureOperators(ctx, 3n, 5n);
+
+    const { chainId } = await ethers.provider.getNetwork();
+    // Sepolia-specific initialization
+    if (chainId === 11155111n) {
```

Comment:
I'd move this to the protocol discovery since this may be used somewhere else, and here, take data from the context or move it under the protocol context flag to explicitly set tests state for Sepolia network.